internal/backup_push_handler.go

```diff
@@ -180,18 +186,34 @@ func HandleBackupPush(uploader *Uploader, archiveDirectory string) {
 	currentBackupSentinelDto.UserData = GetSentinelUserData()
 
 	// If other parts are successful in uploading, upload json file.
-	err = UploadSentinel(uploader, currentBackupSentinelDto, backupName)
+	err = UploadSentinel(uploader, currentBackupSentinelDto, backupName, meta)
 	if err != nil {
 		tracelog.ErrorLogger.Printf("Failed to upload sentinel file for backup: %s", backupName)
 		tracelog.ErrorLogger.FatalError(err)
 	}
 }
 
 // TODO : unit tests
-func UploadSentinel(uploader *Uploader, sentinelDto *BackupSentinelDto, backupName string) error {
+func UploadSentinel(uploader *Uploader, sentinelDto *BackupSentinelDto, backupName string, meta ExtendedMetadataDto) error {
+	meta.FinishTime = time.Now()
```

Comment:
Let's move metadata creation and uploading to a separate function: "UploadMetadata". This will save us from using incomplete metadata object and separate its logic from logic related to Sentinel.